### PR TITLE
unix, windows: Added uv_thread_equal.

### PR DESCRIPTION
**Pull request by neojski**
Implementing uv_thread_equal that compare thread just like pthread_equals.
Changing the return type of uv_thread_self to more proper one uv_thread_t.
Adding simple test.

Fixes issue #1480.

### CODE REVIEW

**Comment by saghul** [comment]
Thanks! Please add the test also to Makefile.am so that it's also built in autoconf builds.

**Comment by neojski** [comment]
I just added test to Makefile.am. Let me know if there are other issues.

**Comment by saghul** [comment]
Any objection @bnoordhuis, @indutny?

File: include/uv.h [hunk 1]
```diff
@@ -2376,8 +2376,9 @@ UV_EXTERN void uv_key_set(uv_key_t* key, void* value);
 typedef void (*uv_thread_cb)(void* arg);
 
 UV_EXTERN int uv_thread_create(uv_thread_t* tid, uv_thread_cb entry, void* arg);
-UV_EXTERN unsigned long uv_thread_self(void);
+UV_EXTERN uv_thread_t uv_thread_self(void);
 UV_EXTERN int uv_thread_join(uv_thread_t *tid);
+UV_EXTERN int uv_thread_equal(uv_thread_t* t1, uv_thread_t* t2);
```
Comment (bnoordhuis) [inline include/uv.h]: The pointers should be const to signal that the inputs are not modified.

File: src/uv-common.c [hunk 2]
```diff
@@ -306,11 +306,11 @@ int uv_thread_create(uv_thread_t *tid, void (*entry)(void *arg), void *arg) {
...
 
 
-unsigned long uv_thread_self(void) {
+uv_thread_t uv_thread_self(void) {
 #ifdef _WIN32
-  return (unsigned long) GetCurrentThreadId();
+  return (uv_thread_t) GetCurrentThreadId();
 #else
-  return (unsigned long) pthread_self();
+  return (uv_thread_t) pthread_self();
```
Comment (bnoordhuis) [inline src/uv-common.c]: The casts aren't necessary.

**Comment by neojski** [comment]
Good points @bnoordhuis. I updated the patch!

File: test/test-thread-equal.c [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include "uv.h"
+#include "task.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h> /* memset */
```
Comment (saghul) [inline test/test-thread-equal.c]: Unneeded?

Comment (neojski) [inline test/test-thread-equal.c]: Yeah, copy-paste artifact. Will fix.

By the way:
test-tcp-close.c and test-thread-equal.c has the same line as 27 above but they don't use memset.

**Comment by neojski** [comment]
Unnecessary includes removed.

**Comment by saghul** [comment]
Thanks @neojski! Landed in 59658a8

**Pull request closed without merging**